website/static/js/waterbutler.js

```diff
@@ -22,67 +21,74 @@ function getDefaultOptions(path, provider) {
     return options;
 }
 
-function buildUrl(suffix, path, provider, nid, options) {
-    path = path || '/';
-    var baseUrl = window.contextVars.waterbutlerURL + suffix;
-    return baseUrl + $.param($.extend(getDefaultOptions(path, provider), {nid: nid}, options));
-}
-
-var buildCrudUrl = buildUrl.bind(this, 'file?');
-var buildCopyUrl = buildUrl.bind(this, 'copy?');
-var buildMoveUrl = buildUrl.bind(this, 'move?');
-var buildMetadataUrl = buildUrl.bind(this, 'data?');
-var buildRevisionsUrl = buildUrl.bind(this, 'revisions?');
-var buildCreateFolderUrl = buildUrl.bind(this, 'folders?');
-
-
-function buildUploadUrl(path, provider, nid, file, options) {
-    path = (path || '/') + file.name;
-    return buildUrl('file?', path, provider, nid, options);
+function getViewOnly() {
+    return $osf.urlParams().view_only;
 }
 
-function buildFromTreebeard(suffix, item, options) {
-    return buildUrl(suffix, item.data.path, item.data.provider, item.data.nodeId, options);
+function buildFromTreebeard(item, options) {
+    return buildUrl(item.data.path, item.data.provider, item.data.nodeId, options);
 }
 
-function buildFromTreebeardFile(item, file, options) {
-    return buildUploadUrl(item.data.path, item.data.provider, item.data.nodeId, file, options);
+function addToken(options) {
+    var wb_options = $.extend({}, options || {}, {token: window.contextVars.accessToken});
+    return wb_options;
 }
 
-function toJsonBlob(item, options) {
-    return $.extend(getDefaultOptions(item.data.path || '/', item.data.provider), {nid: item.data.nodeId}, options);
+// Move obj.data.attributes contents to obj.data , for WB v1 API
+function wbLazyLoadPreprocess(obj) {
+    // WB v1 API now stores file info in data.attributes.
+    // Treebeard expects to find it in data
+    function _promoteAttrs(obj_data) {
+        var saved_attributes = obj_data.attributes;
+        delete obj_data.attributes;
```

Comment:
Please add a comment, `// delete before extending in case saved_attributes has its own attributes property`.